src/Turbo/Broadcaster/TwigMercureBroadcaster.php

```diff
@@ -0,0 +1,151 @@
+        ?ExpressionLanguage $expressionLanguage = null,
+        ?string $entityNamespace = null
+    ) {
+        if (80000 > \PHP_VERSION_ID) {
+            throw new \LogicException('The broadcast feature requires PHP 8.0 or greater, you must either upgrade to PHP 8 or disable it.');
+        }
+
+        if (null === $messageBus && null === $publisher) {
+            throw new \InvalidArgumentException('A message bus or a publisher must be provided.');
+        }
+
+        $this->twig = $twig;
+        $this->propertyAccessor = $propertyAccessor ?? PropertyAccess::createPropertyAccessor();
+        $this->messageBus = $messageBus;
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? new ExpressionLanguage();
+        $this->entityNamespace = $entityNamespace;
+    }
+
+    public function broadcast(object $entity, string $action): void
+    {
+        if (!$attribute = (new \ReflectionClass($entity))->getAttributes(Broadcast::class)[0] ?? null) {
+            return;
+        }
+
+        /**
+         * @var Broadcast $broadcast
+         */
+        $broadcast = $attribute->newInstance();
+        $options = $this->normalizeOptions($entity, $action, $broadcast->options);
+
+        // What must we do if the template or the block doesn't exist? Throwing for now.
+        $data = $this->twig->load($options['template'])->renderBlock($action, ['entity' => $entity, 'action' => $action, 'options' => $options]);
+
+        $update = new Update(
+            $options['topics'],
+            $data,
+            $options['private'] ?? false,
+            $options['id'] ?? null,
+            $options['type'] ?? null,
+            $options['retry'] ?? null
+        );
+
+        $this->messageBus ? $this->messageBus->dispatch($update) : ($this->publisher)($update); // @phpstan-ignore-line
+    }
+
+    /**
+     * @param mixed[] $options
+     *
+     * @return mixed[]
+     */
+    private function normalizeOptions(object $entity, string $action, array $options): array
+    {
+        if (\is_string($options[0] ?? null)) {
+            if (null === $this->expressionLanguage) {
+                throw new \RuntimeException('The Expression Language component is not installed. Try running "composer require symfony/expression-language".');
+            }
+
+            $options = $this->expressionLanguage->evaluate($options[0], ['entity' => $entity, 'action' => $action]);
+        }
+
+        $entityClass = \get_class($entity);
+
+        if ($extraKeys = array_diff(array_keys($options), self::OPTIONS)) {
+            throw new \InvalidArgumentException(sprintf('Unknown broadcast options "%s" on class "%s". Valid options are: "%s"', implode('", "', $extraKeys), $entityClass, implode('", "', self::OPTIONS)));
+        }
+
+        $options['topics'] = (array) ($options['topics'] ?? sprintf(self::TOPIC_PATTERN, rawurlencode($entityClass), rawurlencode($this->propertyAccessor->getValue($entity, 'id'))));
```

Comment:
Should we get the identifier property from the entity class metadata?

Reply:
WDYM? I guess this could be improved in API Platform but not in symfony. What tells you this class is tight to doctrine?

Reply:
I mean that this makes the assumption that the entity class has a property named `id`, which might not be true. 

> What tells you this class is tight to doctrine?

Nothing, indeed. Should the `id` property be configurable then?

Reply:
:+1:

Reply:
It's a rabbit hole. In API Platform we started with this exact same convention, and later we added an option to make the ID configurable, then we added support for composed identifiers, complex identifiers (objects such as dates) etc...
To be honest, I regret having added that complexity to API Platform.

My opinion is that to use this feature, you **must** have a property named `id` or a getter named `getId()` (we could even introduce an interface for that). It's easy, and doable by the user even for the most complex case (even if you use another property as identifier, have complex identifiers etc, you can always add a `getId()` method.

By the way, this mandatory convention is documented in the readme.

WDYT?

Reply:
Works for me, we can revisit later on if needed.